scipy/stats/_distn_infrastructure.py

```diff
@@ -1203,7 +1204,7 @@ def moment(self, n, *args, **kwds):
 
         # Convert to transformed  X = L + S*Y
         # E[X^n] = E[(L+S*Y)^n] = L^n sum(comb(n, k)*(S/L)^k E[Y^k], k=0...n)
-        if loc == 0:
+        if np.all(loc == 0):
```

Comment:
I haven't read the surrounding code, but should this do different things for the elements that are zero vs the ones that aren't zero?

Reply:
Oh, true! sorry for that. I will take a deeper look and add tests soon. Thanks for the review

Reply:
@mdhaber I think I have done it right this time. I can't test it as I am unable to run tests. When I try to run tests using pytest on Linux, I get this error: ``AttributeError: module 'scipy.interpolate.dfitpack' has no attribute 'types'`` at line 38 of ``scipy/interpolate/_fitpack_impl.py``. I built it twice from source and it builds too. Is there anything I am missing?

Reply:
What does `git status` say?
Are you using `runtests.py` from the command line or some other way of running tests?
What instructions did you use to build SciPy?

Reply:
Aha! Using ``runtests.py`` did work! Is there a way to run specific tests only?

Reply:
Thanks! ``runtests.py`` did work!!